### ISSUE TITLE
pageserver: unexpected left-over task in `test_ondemand_download_large_rel`

### ISSUE THREAD

**Issue by jcsp** [neondatabase/neon#5684]
The code by this log says -- so let's investigate this as a bug
```
                    // warn to catch these in tests; there shouldn't be any
```

Job: https://github.com/neondatabase/neon/actions/runs/6657929721/job/18094020427

Allure: https://neon-github-public-dev.s3.amazonaws.com/reports/main/6657929721/index.html#suites/b97efae3a617afb71cb8142f5afa5224/1db8a49eab2cfec3/

Error:
```
2023-10-26T18:13:48.1669061Z ERROR test_runner/regress/test_ondemand_download.py::test_ondemand_download_large_rel[release-pg14-real_s3] - assert not ['2023-10-26T18:08:02.470566Z  WARN shutdown_pageserver{exit_code=0}: stopping left-over name="remote upload" tenant_id=Some(273e26df756e8f8eb5a7c62e1b4bd2dd) timeline_id=Some(2fad4a0c931f2b16092750345b9c3e45) kind=RemoteUploadTask\n']
```

**Comment by koivunej** [comment]
Yes recognized in https://github.com/neondatabase/neon/pull/4938#issuecomment-1780124562 -- however I was thinking the problem had already gone been hidden by deletion queue changes; it just failed to happen for me on the rebase-balooza.

Quoting myself in case the comment is hidden by default:

> > > #### Failures on Posgres 14
> > > 
> > > * `test_ondemand_download_large_rel[local_fs]`: [release](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4938/6012579812/index.html#suites/b97efae3a617afb71cb8142f5afa5224/ab10ec7f4e3c6445)
> > 
> > 
> > This now keeps failing because we never actually stop the remote timeline client, and deletes happen late.
> 
> Fixing this is rather involved, I think we need a new queue operation which we would call from
> 
> https://github.com/neondatabase/neon/blob/85f4514e7d1fb151fe6067a47563657c9df9c76c/pageserver/src/tenant/timeline.rs#L995
>
> like, `wait_for_completion_and_stop`. If anything is even needed because of the https://github.com/neondatabase/neon/pull/4960. I think I'll just leave it not done, and we'll need to remember this if some flakyness arises.

**Comment by koivunej** [comment]
`test_runner/regress/test_tenant_delete.py` and `test_runner/regress/test_timeline_delete.py` are also affected by this nagging in shutdown, but it is not because of `struct Layer` but the deletes which are spawned just late.
